splash/tests/test_execute.py

```diff
@@ -332,6 +332,70 @@ def test_pcall_wrong_keyword_arguments(self):
         self.assertEqual(data["ok"], False)
 
 
+class EnableDisableJSTest(BaseLuaRenderTest):
+
+    def test_disablejs(self):
+        resp = self.request_lua("""
+        function main(splash)
+            splash:disable_js()
+            splash:go(splash.args.url)
+            local html = splash:html()
+            return html
+        end
+        """, {
+            'url': self.mockurl('jsrender'),
+        })
+        self.assertStatusCode(resp, 200)
+        self.assertIn(u'Before', resp.text)
+
+    def test_enablejs(self):
+        resp = self.request_lua("""
+        function main(splash)
+            splash:disable_js()
+            splash:enable_js()
+            splash:go(splash.args.url)
+            local html = splash:html()
+            return html
+        end
+        """, {
+            'url': self.mockurl('jsrender'),
+        })
+        self.assertStatusCode(resp, 200)
+        self.assertIn(u'After', resp.text)
+
+    def test_disablejs_after_splash_go(self):
+        # disable_js doesn't work when called after splash.go(...)
```

Comment:
The test doesn't show that. HTML may have been changed while JS was still enabled, but splash:disable_js() might still disable further JS processing.